### ISSUE TITLE
torchaudio.so: undefined symbol: WavpackGetMode

### ISSUE THREAD

**Issue by rien333** [pytorch/audio#1461]
## 🐛 Bug

Importing torchaudio raises the following error:

```
>>> import torchaudio
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.9/site-packages/torchaudio/__init__.py", line 1, in <module>
    from . import extension  # noqa: F401
  File "/usr/lib/python3.9/site-packages/torchaudio/extension/__init__.py", line 5, in <module>
    _init_extension()
  File "/usr/lib/python3.9/site-packages/torchaudio/extension/extension.py", line 11, in _init_extension
    _init_script_module(ext)
  File "/usr/lib/python3.9/site-packages/torchaudio/extension/extension.py", line 19, in _init_script_module
    torch.classes.load_library(path)
  File "/usr/lib/python3.9/site-packages/torch/_classes.py", line 46, in load_library
    torch.ops.load_library(path)
  File "/usr/lib/python3.9/site-packages/torch/_ops.py", line 104, in load_library
    ctypes.CDLL(path)
  File "/usr/lib/python3.9/ctypes/__init__.py", line 374, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: /usr/lib/python3.9/site-packages/torchaudio/_torchaudio.so: undefined symbol: WavpackGetMode
```

## To Reproduce

1. Obtain the [torchaudio-git](https://aur.archlinux.org/packages/python-torchaudio-git/) PKGBUILD from the AUR, and install. (If you don't run arch linux, the PKGBUILD is just a bunch of bash commands, so it's quite easy to follow along)

3. Run `import torchaudio`

## Expected behavior

I expect the .so file to initialize normally.

## Environment

 - **What commands did you used to install torchaudio (conda/pip/build from source)?**
 see above
 - **If you are building from source, which commit is it?**
 The most current one —https://github.com/pytorch/audio/commit/5cca0008a3f4388d961016e2e06e96206e378d2c
 - **What does `torchaudio.__version__` print? (If applicable)**
 I cannot import the module, unfortunately.

PyTorch version: 1.8.1
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Arch Linux (x86_64)
GCC version: (GCC) 10.2.0
Clang version: 11.1.0
CMake version: version 3.20.1

Python version: 3.9 (64-bit runtime)
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.20.1
[pip3] torch==1.8.1
[pip3] torchaudio==0.9.0a0+5cca000
[conda] Could not collect

## Additional context

Note that other people seem to be having the same problem: https://aur.archlinux.org/packages/python-torchaudio-git/#news

FWIW, I do have `extra/wavpack` from the official Arch Linux repo's installed.

**Comment by mthrok** [comment]
Hi @rien333 

Thanks for letting us know. Looks like this is a configuration issue when building with SoX.

In our testing environments, we do not have `wavpack`, so the build proceeded without `wavpack`. 

```
$ grep -m 1 -A 5 wavpack build/temp.linux-x86_64-3.8/third_party/sox/src/sox-build/config.log
configure:15359: checking wavpack/wavpack.h usability
configure:15359: gcc -c -I/scratch/moto/torchaudio/third_party/sox/../install/include -fvisibility=hidden  -fstack-protector -Wall -W -Wmissing-prototypes -Wstrict-prototypes -pedantic   conftest.c >&5
conftest.c:93:10: fatal error: wavpack/wavpack.h: No such file or directory
 #include <wavpack/wavpack.h>
          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
```

But I think in your environment, the script detected `wavpack` and built `sox` with support for dynamically loading it, but could not resolve its library at the runtime so it ended up `undefined symbol`.

I think the simple fix is to disable it.

```bash
/torchaudio $ ./build/temp.linux-x86_64-3.8/third_party/sox/src/sox/configure --help
<--snip-->
  --with-wavpack=dyn      load wavpack dynamically
<--snip-->
```

I made #1462 for this fix.

**Comment by rien333** [comment]
That may well be it — giving #1462 a try now. Thanks for the quick response!

**Comment by mthrok** [comment]
Merged #1462. Closing the issue. Let us know if the issue persists.
